### PR TITLE
Closes #180 | Implement `IndoMMLU` dataloader

### LINKED ISSUE #180: Create dataset loader for IndoMMLU
Dataloader name: `indommlu/indommlu.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?indommlu

|     Dataset| indommlu |
|-------------|---|
| Description | IndoMMLU is the first multi-task language understanding benchmark for Indonesian culture and languages, which consists of questions from primary school to university entrance exams in Indonesia. By employing professional teachers, we obtained 14,906 questions across 63 tasks and education levels, with 46% of the questions focusing on assessing proficiency in the Indonesian language and knowledge of nine local languages and cultures in Indonesia. |
| Subsets | - |
| Languages | ind |
| Tasks | Question Answering |
| License | Creative Commons Attribution 2.0 (cc-by-2.0) |
| Homepage | https://huggingface.co/datasets/indolem/IndoMMLU |
| HF URL | https://huggingface.co/datasets/indolem/IndoMMLU |
| Paper URL | https://arxiv.org/pdf/2310.04928.pdf |

### PR DESCRIPTION
**Pull request by chenxwh**
Closes #180.

Found 2 invalid examples in the original datasets: the answer is not in the options. The implemented dataloader skips those 2 examples.

```
(4223)
{ 'question': 'Keberadaan jumlah belalang warna hijau dan coklat di sawah pada awalnya seimbang. Setelah padi dipanen, ternyata berpengaruh pada jumlah belalang berwarna hijau yang jumlahnya semakin menurun. Hal ini disebabkan karena ….',
 'options': "['A. belalang berwarna hijau tidak dapat berkembang biak', 'B. jumlahnya berkurang karena dimakan oleh pemangsa', 'C. tidak dapat beradaptasi dengan lingkungan sawah', 'D. adanya pengaruh predasi antara sesama belalang']",
 'answer': 'E'}

(14150)
{'question': 'Ibu Meminta Nana menyapu halaman, tetapi ia sedang belajar, kalimat penolakan yang tetap dalah.....',
 'options': "['A. Aku tidak mau ibu', 'B. Maaf ibu aku sedang belajar', 'C. Ibu saja yang menyapu']",
 'answer': 'D'}
```

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

**Review by holylovenia — CHANGES_REQUESTED** [review]
Hi @chenxwh, thanks for implementing this dataloader. I have some suggestions.

1. Could you please add an empty `__init__.py`?

2. Could you please remove the letters from the answer options?

From this:
```
{'answer': ['C. Seharusnya Ibra tidur lebih awal agar sampai di sekolah tepat waktunya'],
 'choices': ['A. Seharusnya orang tuanya melarang Ibra tidur hingga larut malam',
             'B. Salah sendiri ia tidur larut malam',
             'C. Seharusnya Ibra tidur lebih awal agar sampai di sekolah tepat waktunya',
             'D. Seharusnya Ibra tidak perlu nonton TV hingga larut malam'],
... }
```

Becomes this:

```
{'answer': ['Seharusnya Ibra tidur lebih awal agar sampai di sekolah tepat waktunya'],
 'choices': ['Seharusnya orang tuanya melarang Ibra tidur hingga larut malam',
             'Salah sendiri ia tidur larut malam',
             'Seharusnya Ibra tidur lebih awal agar sampai di sekolah tepat waktunya',
             'Seharusnya Ibra tidak perlu nonton TV hingga larut malam'],
```

3. Could we have more `seacrowd` schemas to separate the indigenous language subjects?

Current subsets:
- `indommlu_source`
- `indommlu_seacrowd_qa`

Suggested subsets:
- `indommlu_source` --> contains everything
- `indommlu_seacrowd_qa` --> contains everything
- `indommlu_ind_seacrowd_qa` --> everything - indigenous language subjects (e.g., Lampungic, Makassarese, Balinese, etc.)
- `indommlu_{lang}_seacrowd_qa` --> for lang in indigenous language subjects (e.g., Lampungic, Makassarese, Balinese, etc.) Example: `indommlu_ban_seacrowd_qa` for Balinese language subject QAs.

4. Update `_LANGUAGES` to contain `ind` as well as all the other indigenous languages in the dataset.

**Comment by chenxwh** [comment]
Thanks for the suggestions! Modified accordingly with added extra langs that have iso code from https://iso639-3.sil.org/code_tables/639/data

File: seacrowd/sea_datasets/indommlu/indommlu.py [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+    "Sosiologi": "Social science",
+    "Ekonomi": "Social science",
+    "Bahasa Sunda": "Local languages and cultures",
+    "Bahasa Jawa": "Local languages and cultures",
+    "PPKN": "Social science",
+}
+
+special_case = ["SD-SMP-SMA", "SD-SMP"]
+level_mapper = {
+    "SMA": "SMA",
```
Comment (holylovenia) [inline seacrowd/sea_datasets/indommlu/indommlu.py:118]: Add this comment: `"# SMA --> high school level"`.

File: seacrowd/sea_datasets/indommlu/indommlu.py [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+    "Seleksi PTN": "University entrance test",
+    "SD": "SD",
```
Comment (holylovenia) [inline seacrowd/sea_datasets/indommlu/indommlu.py:120]: Add this comment: `"# SD --> elementary school level"`.

File: seacrowd/sea_datasets/indommlu/indommlu.py [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+    "SMP": "SMP",
```
Comment (holylovenia) [inline seacrowd/sea_datasets/indommlu/indommlu.py:121]: Add this comment: `"# SMP --> junior high school level"`.

**Review by holylovenia — CHANGES_REQUESTED** [review]
Hi @chenxwh, thanks a lot for the changes. 🙏 Could I please request a few other adjustments?

- `indommlu_ind_seacrowd_qa` to include the Indonesian language subject and other non-language subjects (e.g., History, Math, Geography, etc.)

**Comment by chenxwh** [comment]
Thank you for the comments, I have updated accordingly, apart from adding `"min"` as there is no "Minangkabau language" but only ""Minangkabau culture".

**Comment by holylovenia** [comment]
Replacing @gentaiscool with @ryanignatius due to no response.

**Review by holylovenia — APPROVED** [review]
Tried and tested. Works well! Thanks a lot for the changes and your prompt responses, @chenxwh! 😄

File: seacrowd/sea_datasets/indommlu/indommlu.py [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+    ]
+    for lang in _LANGUAGES:
+        lang_config = SEACrowdConfig(
+            name=f"{_DATASETNAME}_{lang}_seacrowd_qa",
+            version=SEACROWD_VERSION,
+            description=f"{_DATASETNAME} {lang} SEACrowd schema",
+            schema=f"seacrowd_{lang}_qa",
+            subset_id=_DATASETNAME,
+        )
+        BUILDER_CONFIGS.append(lang_config)
```
Comment (ryanignatius) [inline seacrowd/sea_datasets/indommlu/indommlu.py:205]: please also add the source config for all languages
`name=f"{_DATASETNAME}_{lang}_source"`

**Comment by chenxwh** [comment]
oh really? There is no such "subset distinction" in the source datasets. thought @holylovenia suggested we add those just for seacrowd scheme?

**Comment by holylovenia** [comment]
> oh really? There is no such "subset distinction" in the source datasets. thought @holylovenia suggested we add those just for seacrowd scheme?

I think @ryanignatius suggested it so all the subset IDs (i.e., `"indommlu"` and `"indommlu_{lang}"`) can pass the unit test: `python -m tests.test_seacrowd seacrowd/sea_datasets/indommlu/indommlu.py --subset_id={subset_id}`. That being said, I think only having `indommlu_source` as the only source schema is sufficient since all the subsets follow the same schemas.

**Comment by ryanignatius** [comment]
> > oh really? There is no such "subset distinction" in the source datasets. thought @holylovenia suggested we add those just for seacrowd scheme?
> 
> I think @ryanignatius suggested it so all the subset IDs (i.e., `"indommlu"` and `"indommlu_{lang}"`) can pass the unit test: `python -m tests.test_seacrowd seacrowd/sea_datasets/indommlu/indommlu.py --subset_id={subset_id}`. That being said, I think only having `indommlu_source` as the only source schema is sufficient since all the subsets follow the same schemas.

ah okay got it,
sorry I miss the comment to add for seacrowd schema only

**Review by ryanignatius — APPROVED** [review]

**Pull request merged**